File: app/src/main/java/seigneur/gauvain/mycourt/utils/SnackbarHelper.java
package seigneur.gauvain.mycourt.utils;

import android.content.Context;
import android.support.design.widget.Snackbar;
import android.support.v4.view.ViewCompat;
import android.view.ViewGroup;

import seigneur.gauvain.mycourt.R;

//from :https://medium.com/@Tgo1014/creating-googles-new-snackbar-b0fe8db6c0eb
public class SnackbarHelper {

    public static void configSnackbar(Context context, Snackbar snack) {
        addMargins(snack);
        setRoundBordersBg(context, snack);
        ViewCompat.setElevation(snack.getView(), 6f);
    }

    private static void addMargins(Snackbar snack) {
        ViewGroup.MarginLayoutParams params = (ViewGroup.MarginLayoutParams) snack.getView().getLayoutParams();
        params.setMargins(12, 12, 12, 12);
        snack.getView().setLayoutParams(params);
    }

    private static void setRoundBordersBg(Context context, Snackbar snackbar) {
        snackbar.getView().setBackground(context.getDrawable(R.drawable.bg_snackbar));
    }
}

File: app/src/main/java/seigneur/gauvain/mycourt/utils/SnackbarHelper.kt
package seigneur.gauvain.mycourt.utils

import android.content.Context
import com.google.android.material.snackbar.Snackbar
import androidx.core.view.ViewCompat
import android.view.ViewGroup

import seigneur.gauvain.mycourt.R

//from :https://medium.com/@Tgo1014/creating-googles-new-snackbar-b0fe8db6c0eb
object SnackbarHelper {

    fun configSnackbar(context: Context, snack: Snackbar,
                       marginLeft:Int?=12,
                       marginTop:Int?=12,
                       marginRight:Int?=12,
                       marginBottom:Int?=12) {
        addMargins(snack, marginLeft!!, marginTop!!, marginRight!!, marginBottom!!)
        setRoundBordersBg(context, snack)
        ViewCompat.setElevation(snack.view, 6f)
    }

    private fun addMargins(snack: Snackbar,
                           marginLeft:Int,
                           marginTop:Int,
                           marginRight:Int,
                           marginBottom:Int) {
        val params = snack.view.layoutParams as ViewGroup.MarginLayoutParams
        params.setMargins(marginLeft, marginTop, marginRight, marginBottom)
        snack.view.layoutParams = params
    }

    private fun setRoundBordersBg(context: Context, snackbar: Snackbar) {
        snackbar.view.background = context.getDrawable(R.drawable.bg_snackbar)
    }
}